src/core/api/preview.ts

```diff
@@ -0,0 +1,38 @@
+import fetch from 'cross-fetch';
+import { EventId } from 'core/nostr/type';
+
+export interface NotePreview {
+  title: string;
+  content: string;
+  image?: string;
+  authorProfile?: {
+    name: string;
+    picture: string;
+  };
+  created_at: number;
+}
+
+export async function getNotePreview(eventId: EventId) {
+  const timeout = 2000;
+  const url = 'https://nostr-preview.onrender.com' + `/e/${eventId}`;
+  try {
+    const res = await fetchWithTimeout(url, timeout);
```

Comment:
After using fetch in Next.js, I recommend adding ﻿{ cache: 'force-cache' } options to ensure that it is permanently cached. This is because the content for the same eventId should remain unchanged, allowing for long-term caching.

see: https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache